flask_app/models.py

```diff
@@ -95,11 +108,26 @@ class TestMetadata(db.Model):
     metadata_item = db.Column(JSONB)
 
 
-class Error(db.Model):
+class SessionMetadata(db.Model):
 
     id = db.Column(db.Integer, primary_key=True)
+    session_id = db.Column(db.Integer, db.ForeignKey('session.id', ondelete='CASCADE'), index=True)
+    metadata_item = db.Column(JSONB)
+
+
+class TestError(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
     test_id = db.Column(db.Integer, db.ForeignKey('test.id', ondelete='CASCADE'), index=True)
     traceback = db.Column(JSON)
     exception_type = db.Column(db.String(256), index=True)
     exception = db.Column(db.String(256), index=True)
+    timestamp = db.Column(db.Float, default=get_current_time)
+
+
+class SessionError(db.Model):
```

Comment:
So perhaps I misunderstood the question and I apologize. 

Why does this require polymorphism? At no point is there a need to make the error behave like two different errors. We always approach the error from some route (be it test or session, it doesn't matter) and we render it the same way... Why wouldn't a simple pair of many-to-many tables suffice here?